File: teg-back/src/main/java/api/dto/GameDTO.java
package api.dto;

import api.model.GamePlayer;
import api.model.User;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class GameDTO {
    private Long id;
    private String name;
    private Integer maxPlayers;
    private String status;
    private LocalDateTime createdAt;
    private UserDTO createdBy;
    private List<GamePlayerDTO> players;
}

File: teg-front/src/app/core/dto/game.dto.ts
export interface UserDTO {
  id: number;
  username: string;
  email: string;
  // Add other fields as needed
}

export interface CountryTroopDTO {
  country: string;
  troops: number;
}

export interface GamePlayerDTO {
  id: any; // You can type this more strictly if you know the structure
  user: UserDTO;
  color: string;
  turnOrder: number;
  joinedAt: string;
  objective?: string;
  countries?: CountryTroopDTO[];
  // Add other fields as needed
}

export interface GameDTO {
  id: number;
  name: string;
  maxPlayers: number;
  status: string;
  createdAt: string;
  createdBy: UserDTO;
  players: GamePlayerDTO[];
} 